src/components/crossDevice/CrossDeviceLink/index.js

```diff
@@ -77,50 +84,136 @@ class CrossDeviceLink extends Component {
   }
 
   render = () =>
-    this.props.roomId ? <CrossDeviceLinkUI roomId={this.props.roomId} /> : <Spinner />
+    this.props.roomId ? <CrossDeviceLinkUI {...this.props}/> : <Spinner />
 }
 
 class CrossDeviceLinkUI extends Component {
   constructor(props) {
     super(props)
-    this.state = {copySuccess: false}
+    this.state = {
+      copySuccess: false,
+      sending: false,
+      mobileNumber: null,
+      error: {}
+    }
   }
 
+  linkId = `${process.env.BASE_36_VERSION}${this.props.roomId}`
+
+  linkCopiedTimeoutId = null
+
   copyToClipboard = (e) => {
+    e.preventDefault()
     this.textArea.select()
     document.execCommand('copy')
     e.target.focus()
     this.setState({copySuccess: true})
+    this.clearLinkCopiedTimeout()
+    this.linkCopiedTimeoutId = setTimeout(() => {
+      this.setState({copySuccess: false})
+    }, 5000)
+  }
+
+  clearLinkCopiedTimeout = () => {
+    if (this.linkCopiedTimeoutId) {
+      clearTimeout(this.linkCopiedTimeoutId)
+    }
+  }
+
+  updateNumber = (mobileNumber) => {
+    this.setState({mobileNumber})
   }
 
-  render({roomId}) {
-    const mobilePath = `${process.env.BASE_36_VERSION}${roomId}`
-    const mobileUrl = `${process.env.MOBILE_URL}/${mobilePath}`
-    const buttonCopy = this.state.copySuccess ? 'Copied' : 'Copy link'
+  clearPreviousAttempts = () => {
+    if (this.hasInvalidNumberError()) this.clearError()
+    if (this.state.mobileNumber) this.clearNumber()
```

Comment:
Is this line needed?

Reply:
Yes, it is needed. 
If I input a valid number it will be store in the state. After I cancel one digit and I make it invalid, it will allow me to send the message without showing an error. This prevents that from happening